### PR TITLE
Preparation for v0.5.0 release

### PR DESCRIPTION
**Pull request by rasolca**
Draft for v0.5.0 release.

### CODE REVIEW

File: CITATION.cff
Comment (rasolca) [inline CITATION.cff:56]: Need to be updated.

File: CHANGELOG.md
Comment (albestro) [inline CHANGELOG.md:14]:
```suggestion
- Don't communicate in algorithms when using single rank communicators. (#1097)
```
?

File: BUILD.md
Comment (albestro) [inline BUILD.md:24]: We can probably drop this since we have a public version of the spack package. IIRC for the end user it should not be relevant our internal version (even for master), right?

File: BUILD.md
Comment (msimberg) [inline BUILD.md:42]: This isn't true anymore, right? Would it make sense to separate the README changes to a new PR?

File: CHANGELOG.md
Comment (msimberg) [inline CHANGELOG.md:18]:
```suggestion
- Implemented a workaround for hipMemcpyDefault 2D memcpys, due to bugs in HIP. (#1106)
```

File: BUILD.md
Comment (rasolca) [reply BUILD.md]: #1138

File: BUILD.md
Comment (rasolca) [reply BUILD.md]: #1138

File: CHANGELOG.md
Comment (msimberg) [inline CHANGELOG.md:7]: Maybe:
```suggestion
- Introduced an option (*) for forcing contiguous GPU communication buffers. (#1096)
```
?

**Review by msimberg — APPROVED** [review]

**Comment by rasolca** [comment]
Regarding other PRs we might want to include:

- MPI polling
- Include file cleanup
- miniapp as tests
- pika 0.25.0

Anything you would exclude?
Anything else you want to include?

**Comment by msimberg** [comment]
* MPI polling
  * I think this needs more testing (and importantly still needs a lot of cleanup, possibly also on the pika side)
* Include file cleanup
  * This could be nice to include
* miniapp as tests
  * I think if I get to it soon we can include it, but I wouldn't block the release for this
* pika 0.25.0
  * This is CI-only and doesn't make a difference if it's in before or after the release

I think what's there is already enough for a release. I guess https://github.com/eth-cscs/DLA-Future/pull/1016 would be included, but we leave it out of the release notes because it's for internal use?

**Comment by RMeli** [comment]
> I guess https://github.com/eth-cscs/DLA-Future/pull/1016 would be included, but we leave it out of the release notes because it's for internal use?

+1 from me for adding https://github.com/eth-cscs/DLA-Future/pull/1016 (but not including it in the release notes).

**Comment by rasolca** [comment]
> > I guess #1016 would be included, but we leave it out of the release notes because it's for internal use?
> 
> +1 from me for adding #1016 (but not including it in the release notes).

It is included as master will be released, but not relevant for the user, therefore excluded from the release notes.

**Comment by rasolca** [comment]
> pika 0.25.0
> 
>     * This is CI-only and doesn't make a difference if it's in before or after the release

No differences, but we can say that v0.5.0 is tested with pika@0.25.0...

**Comment by rasolca** [comment]
Ready to release after merging #1149

File: CHANGELOG.md
Comment (msimberg) [inline CHANGELOG.md:9]:
```suggestion
- Removed special handling of Intel MKL, as it could lead to broken installations. (#1149)
```

File: DOCUMENTATION.md
Comment (msimberg) [inline DOCUMENTATION.md:6]:
```suggestion
- [Documentation of `v0.5.0`](https://eth-cscs.github.io/DLA-Future/v0.5.0/)
```

**Review by msimberg — CHANGES_REQUESTED** [review]
Looks good to me, only two minor requests for changes.

File: CHANGELOG.md
Comment (RMeli) [inline CHANGELOG.md:11]:
```suggestion
    - Manual installations: the user is responsible to correctly set variables (see [BUILD.md](BUILD.md)).
```

**Review by RMeli — APPROVED** [review]

**Review by msimberg — APPROVED** [review]

**Pull request merged**
